src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -46,6 +46,8 @@ namespace nbl::asset::hlsl::impl
 CHLSLCompiler::CHLSLCompiler(core::smart_refctd_ptr<system::ISystem>&& system)
     : IShaderCompiler(std::move(system))
 {
+    m_defaultIncludeFinder->addSearchPath("", core::make_smart_refctd_ptr<CJITIncludeLoader>());
```

Comment:
you'll need to add the search path (and remove) to the include finder inside `compile`

because the PhysicalDeviceFeatures and Limits need to come from `CHLSLCompiler::SOptions`
https://github.com/Devsh-Graphics-Programming/Nabla/pull/605/files#r1392649532

Reply:
Also the search path is supposed to be `nbl/builtin/hlsl/jit`

Reply:
> you'll need to add the search path (and remove) to the include finder inside `compile`
> 
> because the PhysicalDeviceFeatures and Limits need to come from `CHLSLCompiler::SOptions` https://github.com/Devsh-Graphics-Programming/Nabla/pull/605/files#r1392649532
> 
> P.S. you might have to make a `CHLSLCompiler::CWrappedIncludeFinder` which searches its own paths after the one you use fails

On a second thought, both `CWrappedIncludeFinder` and `CJITIncludeLoader` should move to the `video` namespace, and the wrap with added search path should be constructed by `ILogicalDevice` when compiling shaders on-demand.